src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -551,3 +559,48 @@ public SRAAccession asSRAAccession() {
         return accession;
     }
 }
+
+// TODO: replace this with an InputResource type taking PathSpecifier once this interface is pushed down to htsjdk
+class UriInputResource extends InputResource {
```

Comment:
So I'm not totally sure what we should be doing here.  I'm thinking we might want to actually just add an HtsgetInputResource which specifically handles htsget and nothing else.  Then the of(URI, wrapper, wrapper) factory method can check if the URI is an htsget input and create one of those, or create one of the other existing types.  I think the logic would be:
1.  URI.scheme == htsget -> HtsgetInput
2. IOUtils.createPath succeeds -> PathInput
3. everything else: -> URLInput

The issue is that in GATK we exclusively use path and have special handling for it with the wrapper objects, so your idea of converting to URL and retrying is good and makes a lot of sense, but I think we need to go through Path as our first attempt.  

Maybe we can chat about it if what I'm saying doesn't make sense to you.